src/main/config/simplified_tuning.c

```diff
@@ -65,22 +65,50 @@ static void calculateNewPidValues(pidProfile_t *pidProfile)
 
 static void calculateNewDTermFilterValues(pidProfile_t *pidProfile)
 {
-    pidProfile->dterm_lpf1_dyn_min_hz = constrain(DTERM_LPF1_DYN_MIN_HZ_DEFAULT * pidProfile->simplified_dterm_filter_multiplier / 100, 0, DYN_LPF_MAX_HZ);
-    pidProfile->dterm_lpf1_dyn_max_hz = constrain(DTERM_LPF1_DYN_MAX_HZ_DEFAULT * pidProfile->simplified_dterm_filter_multiplier / 100, 0, DYN_LPF_MAX_HZ);
-    pidProfile->dterm_lpf1_static_hz = constrain(DTERM_LPF1_DYN_MIN_HZ_DEFAULT * pidProfile->simplified_dterm_filter_multiplier / 100, 0, DYN_LPF_MAX_HZ);
-    pidProfile->dterm_lpf2_static_hz = constrain(DTERM_LPF2_HZ_DEFAULT * pidProfile->simplified_dterm_filter_multiplier / 100, 0, LPF_MAX_HZ);
-    pidProfile->dterm_lpf1_type = FILTER_PT1;
-    pidProfile->dterm_lpf2_type = FILTER_PT1;
+    if (pidProfile->dterm_lpf1_dyn_min_hz) {
```

Comment:
Not sure why we should **refuse to apply** some simplified settings if `dterm_lpf1_dyn_min_hz` happens to be 0 - this completely breaks simplified settings support in the firmware.
The same applies for all other changes in this pull request.

This will practically make simplified settings unusable everywhere except for maybe in configurator - if we want this we should be removing simplified settings from the firmware again and go back to making it configurator only.

@blckmn can you please looking into reverting this?

Reply:
The filter slider concept here is different from that in 4.2.

Previously if the user enabled the slider, they would always get:
- lowpass 1 enabled in dynamic mode, and
- lowpass 2 enabled

If the user preferred to use the static mode for lowpass 1, or disabled either lowpass 1 or lowpass 2, the sliders would no longer be of any use, and all adjustments would need to be made manually.

While this returned a reliable outcome when filter sliders were enabled, it dumped people into a fully manual configuration environment if they wanted only one gyro lowpass filter, or if they wanted to use a static lowpass 1 rather than a dynamic.

This is a problem because there are many people using and recommending only single static lowpass filter on gyro, and those people would still like to be able to use the slider to adjust their filtering.  

And also some people are recommending a single biquad on D, for instance.

Many filter configurations in our Presets will be like this.

So the intent here is that the sliders will remain a simplified way to adjust cutoffs, but they still be able to adjust the filters even if some are turned off or if lowpass 1 is static.

Hence the above code permits slider tuning of these kinds of configurations.

From my perspective this is the better way to go.

It will mean that where a Preset is provided with a single gyro or D filter active, the user can still use the slider to adjust the cutoff.  This is a significant usability improvement.

Its worth noting that this also allows a more consistent user experience for testing filters off via OSD changes.  Previously if a user in the field had sliders enabled but turned one filter off, the outcome was unpredictable.  Now it is simply that the filter they turned off will be off. They will still be able to adjust the remaining filter with their slider value and it will still work.  

That's what the above code achieves, and is exactly how we envisage these sliders being used in future.

There is no need to revert this.

Reply:
To answer your question directly....

If `dterm_lpf1_dyn_min_hz` "happens to be zero", it means that the dynamic lowpass 1 option is off because the user explicitly switched it off.  Either they enabled the static option, with the CLI switch, or turned the whole of lowpass 1 off by setting both `dterm_lpf1_dyn_min_hz` to zero and `dterm_lpf1_static_hz` to zero manually eg in CLI or OSD.

If the user starts from defaults and sets `dterm_lpf1_dyn_min_hz` to zero, whether by switching to 'static mode' in Configurator, or by entering '0' in CLI or OSD, there is no point in the slider changing the min or max in this situation.  The min is zero, and max is ignored by the firmware.  So long as the `dterm_lpf1_static_hz` is non-zero, the slider will adjust the `static` mode of lowpass 1. The slider will therefore continue to modulate the cutoff of lowpass 1, regardless of whether static or dynamic mode is active - so long as ONE of those options is active.  This is what we want the slider to do.  

If the user sets *both* `dterm_lpf1_dyn_min_hz` and `dterm_lpf1_static_hz` to zero, the user intent is to turn the whole of lowpass 1 off.  In this case there is no point in the slider multiplying zeroes.  The slider will do nothing because the filter is completely off.  The lowpass 1 filter is off because the user wants it off.  But, if the user has disabled lowpass 1 completely, but still has lowpass 2 enabled, the slider will continue to adjust lowpass 2 as expected.

This is the means by which we can now support these single-filter configurations while retaining the ease of use provided by the sliders.